var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -332,6 +333,9 @@ class Mpich(MpichEnvironmentModifications, AutotoolsPackage, CudaPackage, ROCmPa
 
     depends_on("pmix", when="pmi=pmix")
 
+    # Libevent is required when *vendored* PMIx is used
```

Comment:
```suggestion
    # Libevent may be required when *external* PMIx is used
```
FWIW, if we're using an external PMIx it should already be linked to some libevent. Installing another from Spack shouldn't hurt, but it also shouldn't be necessary 🤷.

Reply:
I agree, but that libevent needs to be a dependency. You run into this if it isn't:
```
  >> 420341    /usr/bin/ld: cannot find -levent_core: No such file or directory
  >> 420342    /usr/bin/ld: cannot find -levent_pthreads: No such file or directory
  >> 420343    make[2]: *** [Makefile:8418: lib/libmpi.la] Error 2
     420344    make[2]: *** Waiting for unfinished jobs....
     420345    libtool: compile:  /lus/lfs1aip1/home/ng-arch/orca2.ng-arch/spack/lib/spack/env/
               nvhpc/nvfortran -Isrc/binding/fortran/use_mpi -Isrc/binding/fortran/use_mpi_f08 
               -c src/binding/fortran/use_mpi_f08/wrappers_f/f08ts.f90 -o src/binding/fortran/u
               se_mpi_f08/wrappers_f/f08ts.o >/dev/null 2>&1
     420346    libtool: compile:  /lus/lfs1aip1/home/ng-arch/orca2.ng-arch/spack/lib/spack/env/
               nvhpc/nvfortran -Isrc/binding/fortran/use_mpi -Isrc/binding/fortran/use_mpi_f08 
               -c src/binding/fortran/use_mpi_f08/wrappers_f/pf08ts.f90 -o src/binding/fortran/
               use_mpi_f08/wrappers_f/pf08ts.o >/dev/null 2>&1
     420347    make[2]: Leaving directory '/lus/lfs1aip1/home/ng-arch/orca2.ng-arch/scratch/spa
               ck-stage/spack-stage-mpich-4.2.3-sce2ipynawjo7bi2cjbwlo7xijgb5bbo/spack-src'
  >> 420348    make[1]: *** [Makefile:19903: all-recursive] Error 1
     420349    make[1]: Leaving directory '/lus/lfs1aip1/home/ng-arch/orca2.ng-arch/scratch/spa
               ck-stage/spack-stage-mpich-4.2.3-sce2ipynawjo7bi2cjbwlo7xijgb5bbo/spack-src'
  >> 420350    make: *** [Makefile:5800: all] Error 2
```

Reply:
I understand it is an issue in your build, but this is not really an MPICH problem. If you paste the output of `ldd libpmix.so` on your system, are the libevent libraries listed with their locations? Or are they not found?

Adding a libevent configuration to this MPICH package is not the right solution because we have no dependency on libevent, only a transitive one thru this external PMIx library.

Reply:
They look fine to me:
```sh
~> ldd /usr/lib64/libpmix.so
	linux-vdso.so.1 (0x0000ffff88d80000)
	libm.so.6 => /lib64/libm.so.6 (0x0000ffff88a90000)
	libutil.so.1 => /lib64/libutil.so.1 (0x0000ffff88a60000)
	libdl.so.2 => /lib64/libdl.so.2 (0x0000ffff88a30000)
	libevent_core-2.1.so.6 => /usr/lib64/libevent_core-2.1.so.6 (0x0000ffff889d0000)
	libevent_pthreads-2.1.so.6 => /usr/lib64/libevent_pthreads-2.1.so.6 (0x0000ffff889a0000)
	libhwloc.so.15 => /usr/lib64/libhwloc.so.15 (0x0000ffff88920000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x0000ffff888e0000)
	libc.so.6 => /lib64/libc.so.6 (0x0000ffff88750000)
	/lib/ld-linux-aarch64.so.1 (0x0000ffff88d90000)
	libpciaccess.so.0 => /usr/lib64/libpciaccess.so.0 (0x0000ffff88720000)
	libxml2.so.2 => /usr/lib64/libxml2.so.2 (0x0000ffff88590000)
	libz.so.1 => /usr/lib64/libz.so.1 (0x0000ffff88550000)
	liblzma.so.5 => /usr/lib64/liblzma.so.5 (0x0000ffff884f0000)
```

Also, the linker command triggered by mpich does specifically use `-levent_core` and `-levent_pthreads` to link libevent. Would that not count as a dependency?

Reply:
Sorry about the wall of text, but here's the link output (default upstream spack, no changes made by me).

`libtool: link: /lus/lfs1aip1/home/ng-arch/orca2.ng-arch/spack/lib/spack/env/nvhpc/nvc -shared  -fPIC -DPIC  src/binding/c/.libs/lib_libmpi_la-c_binding.o src/glue/romio/.libs/lib_libmpi_la-all_romio_symbols.o src/mpi/attr/.libs/lib_libmpi_la-attr_impl.o src/mpi/attr/.libs/lib_libmpi_la-attrutil.o src/mpi/attr/.libs/lib_libmpi_la-dup_fn.o src/mpi/coll/allgather/.libs/lib_libmpi_la-allgather_allcomm_nb.o src/mpi/coll/allgather/.libs/lib_libmpi_la-allgather_intra_recursive_doubling.o src/mpi/coll/allgather/.libs/lib_libmpi_la-allgather_intra_brucks.o src/mpi/coll/allgather/.libs/lib_libmpi_la-allgather_intra_k_brucks.o src/mpi/coll/allgather/.libs/lib_libmpi_la-allgather_intra_recexch.o src/mpi/coll/allgather/.libs/lib_libmpi_la-allgather_intra_ring.o src/mpi/coll/allgather/.libs/lib_libmpi_la-allgather_inter_local_gather_remote_bcast.o src/mpi/coll/allgatherv/.libs/lib_libmpi_la-allgatherv_allcomm_nb.o src/mpi/coll/allgatherv/.libs/lib_libmpi_la-allgatherv_intra_recursive_doubling.o src/mpi/coll/allgatherv/.libs/lib_libmpi_la-allgatherv_intra_brucks.o src/mpi/coll/allgatherv/.libs/lib_libmpi_la-allgatherv_intra_ring.o src/mpi/coll/allgatherv/.libs/lib_libmpi_la-allgatherv_inter_remote_gather_local_bcast.o src/mpi/coll/allreduce/.libs/lib_libmpi_la-allreduce_allcomm_nb.o src/mpi/coll/allreduce/.libs/lib_libmpi_la-allreduce_intra_recursive_doubling.o src/mpi/coll/allreduce/.libs/lib_libmpi_la-allreduce_intra_reduce_scatter_allgather.o src/mpi/coll/allreduce/.libs/lib_libmpi_la-allreduce_intra_smp.o src/mpi/coll/allreduce/.libs/lib_libmpi_la-allreduce_intra_tree.o src/mpi/coll/allreduce/.libs/lib_libmpi_la-allreduce_intra_recexch.o src/mpi/coll/allreduce/.libs/lib_libmpi_la-allreduce_intra_ring.o src/mpi/coll/allreduce/.libs/lib_libmpi_la-allreduce_intra_k_reduce_scatter_allgather.o src/mpi/coll/allreduce/.libs/lib_libmpi_la-allreduce_inter_reduce_exchange_bcast.o src/mpi/coll/alltoall/.libs/lib_libmpi_la-alltoall_allcomm_nb.o src/mpi/coll/alltoall/.libs/lib_libmpi_la-alltoall_intra_pairwise_sendrecv_replace.o src/mpi/coll/alltoall/.libs/lib_libmpi_la-alltoall_intra_brucks.o src/mpi/coll/alltoall/.libs/lib_libmpi_la-alltoall_intra_k_brucks.o src/mpi/coll/alltoall/.libs/lib_libmpi_la-alltoall_intra_scattered.o src/mpi/coll/alltoall/.libs/lib_libmpi_la-alltoall_intra_pairwise.o src/mpi/coll/alltoall/.libs/lib_libmpi_la-alltoall_inter_pairwise_exchange.o src/mpi/coll/alltoallv/.libs/lib_libmpi_la-alltoallv_allcomm_nb.o src/mpi/coll/alltoallv/.libs/lib_libmpi_la-alltoallv_intra_pairwise_sendrecv_replace.o src/mpi/coll/alltoallv/.libs/lib_libmpi_la-alltoallv_intra_scattered.o src/mpi/coll/alltoallv/.libs/lib_libmpi_la-alltoallv_inter_pairwise_exchange.o src/mpi/coll/alltoallw/.libs/lib_libmpi_la-alltoallw_allcomm_nb.o src/mpi/coll/alltoallw/.libs/lib_libmpi_la-alltoallw_intra_pairwise_sendrecv_replace.o src/mpi/coll/alltoallw/.libs/lib_libmpi_la-alltoallw_intra_scattered.o src/mpi/coll/alltoallw/.libs/lib_libmpi_la-alltoallw_inter_pairwise_exchange.o src/mpi/coll/barrier/.libs/lib_libmpi_la-barrier_allcomm_nb.o src/mpi/coll/barrier/.libs/lib_libmpi_la-barrier_intra_k_dissemination.o src/mpi/coll/barrier/.libs/lib_libmpi_la-barrier_intra_recexch.o src/mpi/coll/barrier/.libs/lib_libmpi_la-barrier_intra_smp.o src/mpi/coll/barrier/.libs/lib_libmpi_la-barrier_inter_bcast.o src/mpi/coll/bcast/.libs/lib_libmpi_la-bcast_utils.o src/mpi/coll/bcast/.libs/lib_libmpi_la-bcast_allcomm_nb.o src/mpi/coll/bcast/.libs/lib_libmpi_la-bcast_intra_binomial.o src/mpi/coll/bcast/.libs/lib_libmpi_la-bcast_intra_scatter_recursive_doubling_allgather.o src/mpi/coll/bcast/.libs/lib_libmpi_la-bcast_intra_scatter_ring_allgather.o src/mpi/coll/bcast/.libs/lib_libmpi_la-bcast_intra_smp.o src/mpi/coll/bcast/.libs/lib_libmpi_la-bcast_intra_tree.o src/mpi/coll/bcast/.libs/lib_libmpi_la-bcast_intra_pipelined_tree.o src/mpi/coll/bcast/.libs/lib_libmpi_la-bcast_inter_remote_send_local_bcast.o src/mpi/coll/exscan/.libs/lib_libmpi_la-exscan_allcomm_nb.o src/mpi/coll/exscan/.libs/lib_libmpi_la-exscan_intra_recursive_doubling.o src/mpi/coll/gather/.libs/lib_libmpi_la-gather_allcomm_nb.o src/mpi/coll/gather/.libs/lib_libmpi_la-gather_intra_binomial.o src/mpi/coll/gather/.libs/lib_libmpi_la-gather_inter_linear.o src/mpi/coll/gather/.libs/lib_libmpi_la-gather_inter_local_gather_remote_send.o src/mpi/coll/gatherv/.libs/lib_libmpi_la-gatherv_allcomm_nb.o src/mpi/coll/gatherv/.libs/lib_libmpi_la-gatherv_allcomm_linear.o src/mpi/coll/reduce_scatter/.libs/lib_libmpi_la-reduce_scatter_allcomm_nb.o src/mpi/coll/reduce_scatter/.libs/lib_libmpi_la-reduce_scatter_intra_recursive_halving.o src/mpi/coll/reduce_scatter/.libs/lib_libmpi_la-reduce_scatter_intra_pairwise.o src/mpi/coll/reduce_scatter/.libs/lib_libmpi_la-reduce_scatter_intra_recursive_doubling.o src/mpi/coll/reduce_scatter/.libs/lib_libmpi_la-reduce_scatter_intra_noncommutative.o src/mpi/coll/reduce_scatter/.libs/lib_libmpi_la-reduce_scatter_inter_remote_reduce_local_scatter.o src/mpi/coll/reduce_scatter_block/.libs/lib_libmpi_la-reduce_scatter_block_allcomm_nb.o src/mpi/coll/reduce_scatter_block/.libs/lib_libmpi_la-reduce_scatter_block_intra_recursive_halving.o src/mpi/coll/reduce_scatter_block/.libs/lib_libmpi_la-reduce_scatter_block_intra_pairwise.o src/mpi/coll/reduce_scatter_block/.libs/lib_libmpi_la-reduce_scatter_block_intra_recursive_doubling.o src/mpi/coll/reduce_scatter_block/.libs/lib_libmpi_la-reduce_scatter_block_intra_noncommutative.o src/mpi/coll/reduce_scatter_block/.libs/lib_libmpi_la-reduce_scatter_block_inter_remote_reduce_local_scatter.o src/mpi/coll/reduce/.libs/lib_libmpi_la-reduce_allcomm_nb.o src/mpi/coll/reduce/.libs/lib_libmpi_la-reduce_intra_binomial.o src/mpi/coll/reduce/.libs/lib_libmpi_la-reduce_intra_reduce_scatter_gather.o src/mpi/coll/reduce/.libs/lib_libmpi_la-reduce_intra_smp.o src/mpi/coll/reduce/.libs/lib_libmpi_la-reduce_inter_local_reduce_remote_send.o src/mpi/coll/scan/.libs/lib_libmpi_la-scan_allcomm_nb.o src/mpi/coll/scan/.libs/lib_libmpi_la-scan_intra_recursive_doubling.o src/mpi/coll/scan/.libs/lib_libmpi_la-scan_intra_smp.o src/mpi/coll/scatter/.libs/lib_libmpi_la-scatter_allcomm_nb.o src/mpi/coll/scatter/.libs/lib_libmpi_la-scatter_intra_binomial.o src/mpi/coll/scatter/.libs/lib_libmpi_la-scatter_inter_linear.o src/mpi/coll/scatter/.libs/lib_libmpi_la-scatter_inter_remote_send_local_scatter.o src/mpi/coll/scatterv/.libs/lib_libmpi_la-scatterv_allcomm_nb.o src/mpi/coll/scatterv/.libs/lib_libmpi_la-scatterv_allcomm_linear.o src/mpi/coll/neighbor_allgather/.libs/lib_libmpi_la-neighbor_allgather_allcomm_nb.o src/mpi/coll/neighbor_allgatherv/.libs/lib_libmpi_la-neighbor_allgatherv_allcomm_nb.o src/mpi/coll/neighbor_alltoall/.libs/lib_libmpi_la-neighbor_alltoall_allcomm_nb.o src/mpi/coll/neighbor_alltoallv/.libs/lib_libmpi_la-neighbor_alltoallv_allcomm_nb.o src/mpi/coll/neighbor_alltoallw/.libs/lib_libmpi_la-neighbor_alltoallw_allcomm_nb.o src/mpi/coll/iallgather/.libs/lib_libmpi_la-iallgather_intra_sched_recursive_doubling.o src/mpi/coll/iallgather/.libs/lib_libmpi_la-iallgather_intra_sched_brucks.o src/mpi/coll/iallgather/.libs/lib_libmpi_la-iallgather_intra_sched_ring.o src/mpi/coll/iallgather/.libs/lib_libmpi_la-iallgather_inter_sched_local_gather_remote_bcast.o src/mpi/coll/iallgather/.libs/lib_libmpi_la-iallgather_tsp_brucks.o src/mpi/coll/iallgather/.libs/lib_libmpi_la-iallgather_tsp_recexch.o src/mpi/coll/iallgather/.libs/lib_libmpi_la-iallgather_tsp_ring.o src/mpi/coll/iallgatherv/.libs/lib_libmpi_la-iallgatherv_intra_sched_recursive_doubling.o src/mpi/coll/iallgatherv/.libs/lib_libmpi_la-iallgatherv_intra_sched_brucks.o src/mpi/coll/iallgatherv/.libs/lib_libmpi_la-iallgatherv_intra_sched_ring.o src/mpi/coll/iallgatherv/.libs/lib_libmpi_la-iallgatherv_inter_sched_remote_gather_local_bcast.o src/mpi/coll/iallgatherv/.libs/lib_libmpi_la-iallgatherv_tsp_brucks.o src/mpi/coll/iallgatherv/.libs/lib_libmpi_la-iallgatherv_tsp_recexch.o src/mpi/coll/iallgatherv/.libs/lib_libmpi_la-iallgatherv_tsp_ring.o src/mpi/coll/iallgatherv/.libs/lib_libmpi_la-iallgatherv_utils.o src/mpi/coll/iallreduce/.libs/lib_libmpi_la-iallreduce_intra_sched_naive.o src/mpi/coll/iallreduce/.libs/lib_libmpi_la-iallreduce_intra_sched_reduce_scatter_allgather.o src/mpi/coll/iallreduce/.libs/lib_libmpi_la-iallreduce_intra_sched_recursive_doubling.o src/mpi/coll/iallreduce/.libs/lib_libmpi_la-iallreduce_intra_sched_smp.o src/mpi/coll/iallreduce/.libs/lib_libmpi_la-iallreduce_inter_sched_remote_reduce_local_bcast.o src/mpi/coll/iallreduce/.libs/lib_libmpi_la-iallreduce_tsp_recursive_exchange_common.o src/mpi/coll/iallreduce/.libs/lib_libmpi_la-iallreduce_tsp_recexch.o src/mpi/coll/iallreduce/.libs/lib_libmpi_la-iallreduce_tsp_recexch_reduce_scatter_recexch_allgatherv.o src/mpi/coll/iallreduce/.libs/lib_libmpi_la-iallreduce_tsp_ring.o src/mpi/coll/iallreduce/.libs/lib_libmpi_la-iallreduce_tsp_tree.o src/mpi/coll/iallreduce/.libs/lib_libmpi_la-iallreduce_tsp_auto.o src/mpi/coll/ialltoall/.libs/lib_libmpi_la-ialltoall_intra_sched_inplace.o src/mpi/coll/ialltoall/.libs/lib_libmpi_la-ialltoall_intra_sched_brucks.o src/mpi/coll/ialltoall/.libs/lib_libmpi_la-ialltoall_intra_sched_permuted_sendrecv.o src/mpi/coll/ialltoall/.libs/lib_libmpi_la-ialltoall_intra_sched_pairwise.o src/mpi/coll/ialltoall/.libs/lib_libmpi_la-ialltoall_inter_sched_pairwise_exchange.o src/mpi/coll/ialltoall/.libs/lib_libmpi_la-ialltoall_tsp_brucks.o src/mpi/coll/ialltoall/.libs/lib_libmpi_la-ialltoall_tsp_ring.o src/mpi/coll/ialltoall/.libs/lib_libmpi_la-ialltoall_tsp_scattered.o src/mpi/coll/ialltoallv/.libs/lib_libmpi_la-ialltoallv_intra_sched_inplace.o src/mpi/coll/ialltoallv/.libs/lib_libmpi_la-ialltoallv_intra_sched_blocked.o src/mpi/coll/ialltoallv/.libs/lib_libmpi_la-ialltoallv_inter_sched_pairwise_exchange.o src/mpi/coll/ialltoallv/.libs/lib_libmpi_la-ialltoallv_tsp_blocked.o src/mpi/coll/ialltoallv/.libs/lib_libmpi_la-ialltoallv_tsp_inplace.o src/mpi/coll/ialltoallv/.libs/lib_libmpi_la-ialltoallv_tsp_scattered.o src/mpi/coll/ialltoallw/.libs/lib_libmpi_la-ialltoallw_intra_sched_inplace.o src/mpi/coll/ialltoallw/.libs/lib_libmpi_la-ialltoallw_intra_sched_blocked.o src/mpi/coll/ialltoallw/.libs/lib_libmpi_la-ialltoallw_inter_sched_pairwise_exchange.o src/mpi/coll/ialltoallw/.libs/lib_libmpi_la-ialltoallw_tsp_blocked.o src/mpi/coll/ialltoallw/.libs/lib_libmpi_la-ialltoallw_tsp_inplace.o src/mpi/coll/ibarrier/.libs/lib_libmpi_la-ibarrier_intra_sched_recursive_doubling.o src/mpi/coll/ibarrier/.libs/lib_libmpi_la-ibarrier_inter_sched_bcast.o src/mpi/coll/ibarrier/.libs/lib_libmpi_la-ibarrier_intra_tsp_recexch.o src/mpi/coll/ibarrier/.libs/lib_libmpi_la-ibarrier_intra_tsp_dissem.o src/mpi/coll/ibarrier/.libs/lib_libmpi_la-ibarrier_tsp_auto.o src/mpi/coll/ibcast/.libs/lib_libmpi_la-ibcast_intra_sched_binomial.o src/mpi/coll/ibcast/.libs/lib_libmpi_la-ibcast_intra_sched_scatter_ring_allgather.o src/mpi/coll/ibcast/.libs/lib_libmpi_la-ibcast_intra_sched_scatter_recursive_doubling_allgather.o src/mpi/coll/ibcast/.libs/lib_libmpi_la-ibcast_intra_sched_smp.o src/mpi/coll/ibcast/.libs/lib_libmpi_la-ibcast_inter_sched_flat.o src/mpi/coll/ibcast/.libs/lib_libmpi_la-ibcast_tsp_scatterv_allgatherv.o src/mpi/coll/ibcast/.libs/lib_libmpi_la-ibcast_tsp_scatterv_ring_allgatherv.o src/mpi/coll/ibcast/.libs/lib_libmpi_la-ibcast_tsp_tree.o src/mpi/coll/ibcast/.libs/lib_libmpi_la-ibcast_tsp_auto.o src/mpi/coll/ibcast/.libs/lib_libmpi_la-ibcast_utils.o src/mpi/coll/iexscan/.libs/lib_libmpi_la-iexscan_intra_sched_recursive_doubling.o src/mpi/coll/igather/.libs/lib_libmpi_la-igather_intra_sched_binomial.o src/mpi/coll/igather/.libs/lib_libmpi_la-igather_inter_sched_short.o src/mpi/coll/igather/.libs/lib_libmpi_la-igather_inter_sched_long.o src/mpi/coll/igather/.libs/lib_libmpi_la-igather_tsp_tree.o src/mpi/coll/igatherv/.libs/lib_libmpi_la-igatherv_allcomm_sched_linear.o src/mpi/coll/igatherv/.libs/lib_libmpi_la-igatherv_tsp_linear.o src/mpi/coll/ireduce_scatter/.libs/lib_libmpi_la-ireduce_scatter_intra_sched_recursive_halving.o src/mpi/coll/ireduce_scatter/.libs/lib_libmpi_la-ireduce_scatter_intra_sched_pairwise.o src/mpi/coll/ireduce_scatter/.libs/lib_libmpi_la-ireduce_scatter_intra_sched_recursive_doubling.o src/mpi/coll/ireduce_scatter/.libs/lib_libmpi_la-ireduce_scatter_intra_sched_noncommutative.o src/mpi/coll/ireduce_scatter/.libs/lib_libmpi_la-ireduce_scatter_inter_sched_remote_reduce_local_scatterv.o src/mpi/coll/ireduce_scatter/.libs/lib_libmpi_la-ireduce_scatter_tsp_recexch.o src/mpi/coll/ireduce_scatter_block/.libs/lib_libmpi_la-ireduce_scatter_block_intra_sched_recursive_halving.o src/mpi/coll/ireduce_scatter_block/.libs/lib_libmpi_la-ireduce_scatter_block_intra_sched_pairwise.o src/mpi/coll/ireduce_scatter_block/.libs/lib_libmpi_la-ireduce_scatter_block_intra_sched_recursive_doubling.o src/mpi/coll/ireduce_scatter_block/.libs/lib_libmpi_la-ireduce_scatter_block_intra_sched_noncommutative.o src/mpi/coll/ireduce_scatter_block/.libs/lib_libmpi_la-ireduce_scatter_block_inter_sched_remote_reduce_local_scatterv.o src/mpi/coll/ireduce_scatter_block/.libs/lib_libmpi_la-ireduce_scatter_block_tsp_recexch.o src/mpi/coll/ireduce/.libs/lib_libmpi_la-ireduce_intra_sched_binomial.o src/mpi/coll/ireduce/.libs/lib_libmpi_la-ireduce_intra_sched_reduce_scatter_gather.o src/mpi/coll/ireduce/.libs/lib_libmpi_la-ireduce_intra_sched_smp.o src/mpi/coll/ireduce/.libs/lib_libmpi_la-ireduce_inter_sched_local_reduce_remote_send.o src/mpi/coll/ireduce/.libs/lib_libmpi_la-ireduce_tsp_tree.o src/mpi/coll/ireduce/.libs/lib_libmpi_la-ireduce_tsp_auto.o src/mpi/coll/iscan/.libs/lib_libmpi_la-iscan_intra_sched_recursive_doubling.o src/mpi/coll/iscan/.libs/lib_libmpi_la-iscan_intra_sched_smp.o src/mpi/coll/iscan/.libs/lib_libmpi_la-iscan_tsp_recursive_doubling.o src/mpi/coll/iscatter/.libs/lib_libmpi_la-iscatter_intra_sched_binomial.o src/mpi/coll/iscatter/.libs/lib_libmpi_la-iscatter_inter_sched_linear.o src/mpi/coll/iscatter/.libs/lib_libmpi_la-iscatter_inter_sched_remote_send_local_scatter.o src/mpi/coll/iscatter/.libs/lib_libmpi_la-iscatter_tsp_tree.o src/mpi/coll/iscatterv/.libs/lib_libmpi_la-iscatterv_allcomm_sched_linear.o src/mpi/coll/iscatterv/.libs/lib_libmpi_la-iscatterv_tsp_linear.o src/mpi/coll/ineighbor_allgather/.libs/lib_libmpi_la-ineighbor_allgather_allcomm_sched_linear.o src/mpi/coll/ineighbor_allgather/.libs/lib_libmpi_la-ineighbor_allgather_tsp_linear.o src/mpi/coll/ineighbor_allgatherv/.libs/lib_libmpi_la-ineighbor_allgatherv_allcomm_sched_linear.o src/mpi/coll/ineighbor_allgatherv/.libs/lib_libmpi_la-ineighbor_allgatherv_tsp_linear.o src/mpi/coll/ineighbor_alltoall/.libs/lib_libmpi_la-ineighbor_alltoall_allcomm_sched_linear.o src/mpi/coll/ineighbor_alltoall/.libs/lib_libmpi_la-ineighbor_alltoall_tsp_linear.o src/mpi/coll/ineighbor_alltoallv/.libs/lib_libmpi_la-ineighbor_alltoallv_allcomm_sched_linear.o src/mpi/coll/ineighbor_alltoallv/.libs/lib_libmpi_la-ineighbor_alltoallv_tsp_linear.o src/mpi/coll/ineighbor_alltoallw/.libs/lib_libmpi_la-ineighbor_alltoallw_allcomm_sched_linear.o src/mpi/coll/ineighbor_alltoallw/.libs/lib_libmpi_la-ineighbor_alltoallw_tsp_linear.o src/mpi/coll/op/.libs/lib_libmpi_la-op_impl.o src/mpi/coll/op/.libs/lib_libmpi_la-oputil.o src/mpi/coll/op/.libs/lib_libmpi_la-opsum.o src/mpi/coll/op/.libs/lib_libmpi_la-opmax.o src/mpi/coll/op/.libs/lib_libmpi_la-opmin.o src/mpi/coll/op/.libs/lib_libmpi_la-opband.o src/mpi/coll/op/.libs/lib_libmpi_la-opbor.o src/mpi/coll/op/.libs/lib_libmpi_la-opbxor.o src/mpi/coll/op/.libs/lib_libmpi_la-opland.o src/mpi/coll/op/.libs/lib_libmpi_la-oplor.o src/mpi/coll/op/.libs/lib_libmpi_la-oplxor.o src/mpi/coll/op/.libs/lib_libmpi_la-opprod.o src/mpi/coll/op/.libs/lib_libmpi_la-opminloc.o src/mpi/coll/op/.libs/lib_libmpi_la-opmaxloc.o src/mpi/coll/op/.libs/lib_libmpi_la-opno_op.o src/mpi/coll/op/.libs/lib_libmpi_la-opreplace.o src/mpi/coll/op/.libs/lib_libmpi_la-opequal.o src/mpi/coll/reduce_local/.libs/lib_libmpi_la-reduce_local.o src/mpi/coll/allreduce_group/.libs/lib_libmpi_la-allreduce_group.o src/mpi/coll/src/.libs/lib_libmpi_la-coll_impl.o src/mpi/coll/src/.libs/lib_libmpi_la-csel.o src/mpi/coll/src/.libs/lib_libmpi_la-csel_container.o src/mpi/coll/src/.libs/lib_libmpi_la-csel_json_autogen.o src/mpi/coll/transports/gentran/.libs/lib_libmpi_la-gentran_impl.o src/mpi/coll/transports/gentran/.libs/lib_libmpi_la-gentran_utils.o src/mpi/coll/transports/gentran/.libs/lib_libmpi_la-tsp_gentran.o src/mpi/coll/algorithms/treealgo/.libs/lib_libmpi_la-treealgo.o src/mpi/coll/algorithms/treealgo/.libs/lib_libmpi_la-treeutil.o src/mpi/coll/algorithms/recexchalgo/.libs/lib_libmpi_la-recexchalgo.o src/mpi/coll/algorithms/stubalgo/.libs/lib_libmpi_la-stubalgo.o src/mpi/coll/.libs/lib_libmpi_la-mpir_coll_sched_auto.o src/mpi/coll/.libs/lib_libmpi_la-mpir_coll.o src/mpi/coll/.libs/lib_libmpi_la-helper_fns.o src/mpi/coll/.libs/lib_libmpi_la-nbcutil.o src/mpi/comm/.libs/lib_libmpi_la-comm_impl.o src/mpi/comm/.libs/lib_libmpi_la-comm_split.o src/mpi/comm/.libs/lib_libmpi_la-comm_split_type.o src/mpi/comm/.libs/lib_libmpi_la-comm_split_type_nbhd.o src/mpi/comm/.libs/lib_libmpi_la-ulfm_impl.o src/mpi/comm/.libs/lib_libmpi_la-builtin_comms.o src/mpi/comm/.libs/lib_libmpi_la-commutil.o src/mpi/comm/.libs/lib_libmpi_la-contextid.o src/mpi/datatype/typerep/src/.libs/lib_libmpi_la-typerep_ext32.o src/mpi/datatype/typerep/src/.libs/lib_libmpi_la-typerep_flatten.o src/mpi/datatype/typerep/src/.libs/lib_libmpi_la-typerep_op.o src/mpi/datatype/typerep/src/.libs/lib_libmpi_la-typerep_yaksa_create.o src/mpi/datatype/typerep/src/.libs/lib_libmpi_la-typerep_yaksa_init.o src/mpi/datatype/typerep/src/.libs/lib_libmpi_la-typerep_yaksa_pack.o src/mpi/datatype/typerep/src/.libs/lib_libmpi_la-typerep_yaksa_pack_external.o src/mpi/datatype/typerep/src/.libs/lib_libmpi_la-typerep_yaksa_iov.o src/mpi/datatype/typerep/src/.libs/lib_libmpi_la-typerep_yaksa_commit.o src/mpi/datatype/typerep/src/.libs/lib_libmpi_la-typerep_yaksa_debug.o src/mpi/datatype/.libs/lib_libmpi_la-datatype_impl.o src/mpi/datatype/.libs/lib_libmpi_la-typeutil.o src/mpi/datatype/.libs/lib_libmpi_la-type_contents.o src/mpi/datatype/.libs/lib_libmpi_la-get_elements_x.o src/mpi/datatype/.libs/lib_libmpi_la-type_create.o src/mpi/datatype/.libs/lib_libmpi_la-type_create_darray.o src/mpi/datatype/.libs/lib_libmpi_la-type_create_subarray.o src/mpi/datatype/.libs/lib_libmpi_la-type_create_pairtype.o src/mpi/datatype/.libs/lib_libmpi_la-create_f90.o src/mpi/datatype/.libs/lib_libmpi_la-type_debug.o src/mpi/errhan/.libs/lib_libmpi_la-errhan_impl.o src/mpi/errhan/.libs/lib_libmpi_la-errhan_file.o src/mpi/errhan/.libs/lib_libmpi_la-errutil.o src/mpi/errhan/.libs/lib_libmpi_la-dynerrutil.o src/mpi/group/.libs/lib_libmpi_la-group_impl.o src/mpi/group/.libs/lib_libmpi_la-grouputil.o src/mpi/info/.libs/lib_libmpi_la-info_impl.o src/mpi/info/.libs/lib_libmpi_la-infoutil.o src/mpi/init/.libs/lib_libmpi_la-init_impl.o src/mpi/init/.libs/lib_libmpi_la-init_util.o src/mpi/init/.libs/lib_libmpi_la-mpir_init.o src/mpi/init/.libs/lib_libmpi_la-globals.o src/mpi/init/.libs/lib_libmpi_la-initinfo.o src/mpi/init/.libs/lib_libmpi_la-local_proc_attrs.o src/mpi/init/.libs/lib_libmpi_la-mutex.o src/mpi/init/.libs/lib_libmpi_la-init_async.o src/mpi/init/.libs/lib_libmpi_la-init_windows.o src/mpi/init/.libs/lib_libmpi_la-init_bindings.o src/mpi/init/.libs/lib_libmpi_la-init_dbg_logging.o src/mpi/init/.libs/lib_libmpi_la-init_gpu.o src/mpi/misc/.libs/lib_libmpi_la-utils.o src/mpi/misc/.libs/lib_libmpi_la-f2c_impl.o src/mpi/misc/.libs/lib_libmpi_la-memory_alloc_kinds.o src/mpi/pt2pt/.libs/lib_libmpi_la-sendrecv.o src/mpi/pt2pt/.libs/lib_libmpi_la-bsendutil.o src/mpi/request/.libs/lib_libmpi_la-request_impl.o src/mpi/request/.libs/lib_libmpi_la-mpir_greq.o src/mpi/request/.libs/lib_libmpi_la-mpir_request.o src/mpi/rma/.libs/lib_libmpi_la-rma_impl.o src/mpi/rma/.libs/lib_libmpi_la-winutil.o src/mpi/rma/.libs/lib_libmpi_la-rmatypeutil.o src/mpi/session/.libs/lib_libmpi_la-session_impl.o src/mpi/session/.libs/lib_libmpi_la-session_util.o src/mpi/spawn/.libs/lib_libmpi_la-spawn_impl.o src/mpi/topo/.libs/lib_libmpi_la-dims_create.o src/mpi/topo/.libs/lib_libmpi_la-dist_graph_create.o src/mpi/topo/.libs/lib_libmpi_la-topo_impl.o src/mpi/topo/.libs/lib_libmpi_la-topoutil.o src/mpi/stream/.libs/lib_libmpi_la-stream_impl.o src/mpi/stream/.libs/lib_libmpi_la-stream_enqueue.o src/mpi/stream/.libs/lib_libmpi_la-stream_util.o src/mpi/threadcomm/.libs/lib_libmpi_la-threadcomm_impl.o src/mpi/threadcomm/.libs/lib_libmpi_la-threadcomm_pt2pt_impl.o src/mpi/threadcomm/.libs/lib_libmpi_la-threadcomm_coll_impl.o src/util/.libs/lib_libmpi_la-mpir_assert.o src/util/.libs/lib_libmpi_la-mpir_cvars.o src/util/.libs/lib_libmpi_la-mpir_pmi.o src/util/.libs/lib_libmpi_la-mpir_nodemap.o src/util/.libs/lib_libmpi_la-mpir_handlemem.o src/util/.libs/lib_libmpi_la-mpir_strerror.o src/util/.libs/lib_libmpi_la-mpir_localproc.o src/util/.libs/lib_libmpi_la-mpir_netloc.o src/util/.libs/lib_libmpi_la-mpir_hwtopo.o src/util/.libs/lib_libmpi_la-mpir_nettopo.o src/util/.libs/lib_libmpi_la-mpir_progress_hook.o src/glue/romio/.libs/lib_libmpi_la-glue_romio.o src/mpid/ch4/src/.libs/lib_libmpi_la-ch4_globals.o src/mpid/ch4/src/.libs/lib_libmpi_la-ch4_impl.o src/mpid/ch4/src/.libs/lib_libmpi_la-ch4_init.o src/mpid/ch4/src/.libs/lib_libmpi_la-init_comm.o src/mpid/ch4/src/.libs/lib_libmpi_la-ch4_comm.o src/mpid/ch4/src/.libs/lib_libmpi_la-ch4_spawn.o src/mpid/ch4/src/.libs/lib_libmpi_la-ch4_win.o src/mpid/ch4/src/.libs/lib_libmpi_la-ch4_part.o src/mpid/ch4/src/.libs/lib_libmpi_la-ch4_self.o src/mpid/ch4/src/.libs/lib_libmpi_la-ch4i_comm.o src/mpid/ch4/src/.libs/lib_libmpi_la-ch4_proc.o src/mpid/ch4/src/.libs/lib_libmpi_la-ch4_stream_enqueue.o src/mpid/ch4/src/.libs/lib_libmpi_la-ch4_persist.o src/mpid/ch4/src/.libs/lib_libmpi_la-mpidig_init.o src/mpid/ch4/src/.libs/lib_libmpi_la-mpidig_recvq.o src/mpid/ch4/src/.libs/lib_libmpi_la-mpidig_pt2pt_callbacks.o src/mpid/ch4/src/.libs/lib_libmpi_la-mpidig_rma_callbacks.o src/mpid/ch4/src/.libs/lib_libmpi_la-mpidig_win.o src/mpid/ch4/src/.libs/lib_libmpi_la-mpidig_part.o src/mpid/ch4/src/.libs/lib_libmpi_la-mpidig_part_callbacks.o src/mpid/ch4/src/.libs/lib_libmpi_la-mpidig_comm_abort.o src/mpid/ch4/src/.libs/lib_libmpi_la-mpid_ch4_net_array.o src/mpid/ch4/netmod/src/.libs/lib_libmpi_la-netmod_impl.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-func_table.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-ofi_init.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-ofi_comm.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-ofi_datatype.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-ofi_op.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-ofi_rma.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-ofi_spawn.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-ofi_win.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-ofi_part.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-ofi_events.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-ofi_huge.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-ofi_progress.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-ofi_am_events.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-ofi_nic.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-globals.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-init_provider.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-init_settings.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-init_addrxchg.o src/mpid/ch4/netmod/ofi/.libs/lib_libmpi_la-util.o src/mpid/ch4/shm/src/.libs/lib_libmpi_la-shm_init.o src/mpid/ch4/shm/src/.libs/lib_libmpi_la-shm_hooks.o src/mpid/ch4/shm/src/.libs/lib_libmpi_la-shm_mem.o src/mpid/ch4/shm/src/.libs/lib_libmpi_la-shm_rma.o src/mpid/ch4/shm/src/.libs/lib_libmpi_la-shm_part.o src/mpid/ch4/shm/src/.libs/lib_libmpi_la-topotree.o src/mpid/ch4/shm/src/.libs/lib_libmpi_la-topotree_util.o src/mpid/ch4/shm/posix/release_gather/.libs/lib_libmpi_la-release_gather.o src/mpid/ch4/shm/posix/release_gather/.libs/lib_libmpi_la-nb_release_gather.o src/mpid/ch4/shm/posix/.libs/lib_libmpi_la-globals.o src/mpid/ch4/shm/posix/.libs/lib_libmpi_la-posix_comm.o src/mpid/ch4/shm/posix/.libs/lib_libmpi_la-posix_init.o src/mpid/ch4/shm/posix/.libs/lib_libmpi_la-posix_op.o src/mpid/ch4/shm/posix/.libs/lib_libmpi_la-posix_datatype.o src/mpid/ch4/shm/posix/.libs/lib_libmpi_la-posix_win.o src/mpid/ch4/shm/posix/.libs/lib_libmpi_la-posix_part.o src/mpid/ch4/shm/posix/.libs/lib_libmpi_la-posix_eager_array.o src/mpid/ch4/shm/posix/eager/src/.libs/lib_libmpi_la-posix_eager_impl.o src/mpid/ch4/shm/posix/eager/iqueue/.libs/lib_libmpi_la-func_table.o src/mpid/ch4/shm/posix/eager/iqueue/.libs/lib_libmpi_la-iqueue_init.o src/mpid/ch4/shm/ipc/src/.libs/lib_libmpi_la-globals.o src/mpid/ch4/shm/ipc/src/.libs/lib_libmpi_la-ipc_init.o src/mpid/ch4/shm/ipc/src/.libs/lib_libmpi_la-ipc_control.o src/mpid/ch4/shm/ipc/src/.libs/lib_libmpi_la-ipc_win.o src/mpid/ch4/shm/ipc/src/.libs/lib_libmpi_la-ipc_fd.o src/mpid/ch4/shm/ipc/xpmem/.libs/lib_libmpi_la-xpmem_stub.o src/mpid/ch4/shm/ipc/gpu/.libs/lib_libmpi_la-gpu_post.o src/mpid/ch4/shm/ipc/gpu/.libs/lib_libmpi_la-globals.o src/mpid/ch4/shm/ipc/gpu/.libs/lib_libmpi_la-gpu_init.o src/mpid/common/sched/.libs/lib_libmpi_la-mpidu_sched.o src/mpid/common/shm/.libs/lib_libmpi_la-mpidu_init_shm_alloc.o src/mpid/common/shm/.libs/lib_libmpi_la-mpidu_init_shm.o src/mpid/common/shm/.libs/lib_libmpi_la-mpidu_shm_alloc.o src/mpid/common/bc/.libs/lib_libmpi_la-mpidu_bc.o src/mpid/common/genq/.libs/lib_libmpi_la-mpidu_genq_shmem_pool.o src/mpid/common/genq/.libs/lib_libmpi_la-mpidu_genq_shmem_queue.o src/mpid/common/genq/.libs/lib_libmpi_la-mpidu_genq_private_pool.o src/mpid/common/stream_workq/.libs/lib_libmpi_la-stream_workq.o src/mpi_t/.libs/lib_libmpi_la-mpit_finalize.o src/mpi_t/.libs/lib_libmpi_la-mpit_initthread.o src/mpi_t/.libs/lib_libmpi_la-mpit_impl.o src/mpi_t/.libs/lib_libmpi_la-pvar_impl.o src/mpi_t/.libs/lib_libmpi_la-mpit.o src/mpi_t/.libs/lib_libmpi_la-events_impl.o src/mpi_t/.libs/lib_libmpi_la-qmpi_register.o src/nameserv/pmi/.libs/lib_libmpi_la-pmi_nameserv.o  -Wl,--whole-archive src/mpl/.libs/libmpl.a modules/json-c/.libs/libjson-c.a src/mpi/romio/.libs/libromio.a -Wl,--no-whole-archive  -L/opt/cray/libfabric/1.15.2.0/lib -L/opt/cray/libfabric/1.15.2.0/lib64 -L/lus/lfs1aip1/home/ng-arch/orca2.ng-arch/spack/opt/spack/linux-sles15-neoverse_v2/gcc-12.3.0/hwloc-2.11.1-74oldb3ymzgcqngmtniv6ajdvjsbgf5r/lib -L/usr/lib -L/usr/lib64 -L/lus/lfs1aip1/home/ng-arch/orca2.ng-arch/spack/opt/spack/linux-sles15-neoverse_v2/gcc-12.3.0/yaksa-0.3-7n6dtbipjfe4vf5iudtwan6uaghuwr55/lib -L/lus/lfs1aip1/home/ng-arch/orca2.ng-arch/spack/opt/spack/linux-sles15-neoverse_v2/gcc-12.3.0/cuda-12.8.0-ecjqlw5ebpwv3iaute3dnwkogle4dyj6/lib -L/lus/lfs1aip1/home/ng-arch/orca2.ng-arch/spack/opt/spack/linux-sles15-neoverse_v2/gcc-12.3.0/cuda-12.8.0-ecjqlw5ebpwv3iaute3dnwkogle4dyj6/lib64 /usr/lib64/libpmix.so -lm -lutil -levent_core -levent_pthreads -lhwloc -lyaksa -lfabric /usr/lib64/libcxi.so -lnl-3 -latomic -ldl -lpthread -lcudart -lstdc++ -lcuda -lrt  -O2   -pthread -Wl,-soname -Wl,libmpi.so.12 -o lib/.libs/libmpi.so.12.4.3`

Reply:
They are a dependency of a dependency, and there may be other external PMIx libraries that do not depend on libevent. Sorry for being pedantic, but I don't want to make MPICH users think we have some integration with libevent when we do not.

What I suspect is happening is that `-levent_core -levent_pthreads` are getting added by some libtool or pkgconfig file, but your system does not have the dev(el) packages for libevent installed so the linker cannot find them. If you can, try installing `libevent-devel` or similar on your system and see if the linking issue goes away.

Reply:
I can't do this on a national supercomputer unfortunately. It's fine, I'll maintain my own branch for the Isambard Cray systems with patches. Thanks for taking a look.